examples/spreadsheet.md

```diff
@@ -191,20 +234,80 @@ table.addEventListener("keyup", (event) => {
         highlight(target);
     }
 });
+
+table.addEventListener("keydown", (event) => {
+    const target = document.activeElement;
+    switch (event.keyCode) {
+        // tab
+        case 9:
+            event.preventDefault();
+            if (event.shiftKey) {
+                moveSelection(target, -1, 0);
+            } else {
+                moveSelection(target, 1, 0);
+            }
+            break;
+        // left arrow
+        case 37:
+            moveSelection(target, -1, 0);
+            break;
+        // up arrow
+        case 38:
+            moveSelection(target, 0, -1);
+            break;
+        // right arrow
+        case 39:
+            moveSelection(target, 1, 0);
+            break;
+        // down arrow
+        case 40:
+            moveSelection(target, 0, 1);
+            break;
+    }
+});
 ```
 
-This also makes use of `increment()`, which uses some simple metadata-math
-to look up the cell in the next row of this column, and `focus()` it.
+These key handlers also make use of `moveSelection()`, which uses some
+simple metadata-math to look up the next cell in either the `x` or `y`
+direction and update the `SELECTED_POSITION` - scrolling the table if 
+necessary and providing a small buffer to the edge of the visible table.
 
 ```javascript
-function increment(active_cell) {
+const SCROLL_AHEAD = 4;
+
+function moveSelection(active_cell, dx, dy) {
```

Comment:
Awesome work!

What if we could simplify this function by only calling the scrollApi function in only one place? 

I thought of something like this, but it might be worth to check not to be introducing some issues and i'm pretty sure it could be simplified even more:

`
function moveSelection(active_cell, dx, dy) {
    const meta = table.getMeta(active_cell);
    SELECTED_POSITION.x = Math.max(Math.min(meta.x + dx, NUM_COLUMNS - 1), 0);
    SELECTED_POSITION.y = Math.max(Math.min(meta.y + dy, NUM_ROWS - 1), 0);
    const VISIBLE_COLUMNS = Array.from(document.querySelectorAll("tbody tr:first-child td"));
    const VISIBLE_ROWS = Array.from(document.querySelectorAll("tbody tr > td:nth-child(2)"));
    const FIRST_VISIBLE_X = table.getMeta(VISIBLE_COLUMNS[0]).x;
    const FIRST_VISIBLE_Y = table.getMeta(VISIBLE_ROWS[0]).y;
    const LAST_VISIBLE_X = table.getMeta(VISIBLE_COLUMNS.slice(-1)[0]).x;
    const LAST_VISIBLE_Y = table.getMeta(VISIBLE_ROWS.slice(-1)[0]).y;

    let newX0 = meta.x0;
    let newY0 = meta.y0;
    if (
        (meta.x1 < NUM_COLUMNS && SELECTED_POSITION.x > LAST_VISIBLE_X - SCROLL_AHEAD) ||
        SELECTED_POSITION.x - FIRST_VISIBLE_X <= SCROLL_AHEAD ||
        LAST_VISIBLE_X - SELECTED_POSITION.x <= SCROLL_AHEAD
    ) {
        newX0 = Math.max(Math.min(meta.x0 + dx, NUM_COLUMNS - VISIBLE_COLUMNS.length), 0);
    }
    if (
        (meta.y1 < NUM_ROWS && SELECTED_POSITION.y > LAST_VISIBLE_Y - SCROLL_AHEAD) ||
        SELECTED_POSITION.y - FIRST_VISIBLE_Y <= SCROLL_AHEAD ||
        LAST_VISIBLE_Y - SELECTED_POSITION.y <= SCROLL_AHEAD
    ) {
        newY0 = Math.max(Math.min(meta.y0 + dy, NUM_ROWS - VISIBLE_ROWS.length), 0);
    }
    table.scrollTo(newX0, newY0, NUM_COLUMNS, NUM_ROWS);
    updateFocus();
}
`